test/lan_udev_auto_test.rb

```diff
@@ -6,105 +6,135 @@
 
 Yast.import "LanUdevAuto"
 
-describe "#getDeviceName" do
-  NEW_STYLE_NAME = "spec0"
-  MAC_BASED_NAME = "spec-id-00:11:22:33:44:FF"
-  BUS_BASED_NAME = "spec-bus-0000:00:19.0"
-
-  LCASE_MAC_NAME = "spec-id-00:11:22:33:44:ff"
-
-  UNKNOWN_MAC_NAME = "spec-id-00:00:00:00:00:00"
-  UNKNOWN_BUS_NAME = "spec-bus-0000:00:00.0"
-
-  INVALID_NAME = "some funny string"
-
-  # general mocking stuff is placed here
-  before(:each) do
-    # mock devices configuration
-    allow(Yast::LanUdevAuto).to receive(:ReadHardware) {
-      [
-        {
-          "dev_name" => NEW_STYLE_NAME,
-          "mac"      => "00:11:22:33:44:FF",
-          "busid"    => "0000:00:19.0"
-        }
-      ]
-    }
+describe Yast::LanUdevAuto do
+  let(:udev_rules) do
+    [
+      {
+        "rule"  => "ATTR{address}",
+        "value" => "aa:BB:cc:DD:ee:FF",
+        "name"  => "custom-name"
+      }
+    ]
   end
 
-  context "when new style name is provided" do
-    it "returns the new style name" do
-      expect(Yast::LanUdevAuto.getDeviceName(NEW_STYLE_NAME)).to be_equal NEW_STYLE_NAME
+  describe ".getDeviceName" do
+    subject(:getDeviceName) { Yast::LanUdevAuto.getDeviceName(oldname) }
+
+    let(:new_style_name) { "spec0" }
+    let(:mac) { "00:11:22:33:44:FF" }
+    let(:busid) { "0000:00:19.0" }
+
+    # general mocking stuff is placed here
+    before(:each) do
```

Comment:
not blocker just `nth`, as before(:all) is almost not used, so each is good and intuitive default